### ISSUE TITLE
The real cause of RPC request invalidity is concealed

### ISSUE THREAD

**Issue by FabijanC** [0xSpaceShard/starknet-devnet-rs#229]
Users often just get to see that their request is invalid, without the real reason why. E.g as reported in #179
```json
{
  "jsonrpc": "2.0",
  "id": 0,
  "error": {
    "code": -32602,
    "message": "data did not match any variant of untagged enum BroadcastedDeclareTransaction"
  }
}
```

To discover which exactly property was invalid, I had to compare the specification and the request made by the user.

Can this be improved? In the sense of being more specific about what is invalid?

**Comment by marioiordanov** [comment]
no, because untagged enums work like that: https://serde.rs/enum-representations.html#untagged

**Comment by FabijanC** [comment]
I closed the issue, but I see you edited the comment to remove the "no, because..." part. Should it be reopened?

**Comment by marioiordanov** [comment]
I saw that there is a tag that can be used to indicate which object should be serialized.

**Comment by FabijanC** [comment]
Useful: https://www.gustavwengel.dk/serde-untagged-enum-errors-are-bad
